### PR TITLE
Fix potential ZwMapViewOfSection and GetKeyboardLayoutList bugs and add support for SystemCodeInformation class

### PR DESCRIPTION
**Pull request by stonerhash**
Modified ZwMapViewOfSection in order to use win_perms_to_emu_perms to successfully map Windows permission to emulation engine permissions. Definition of win_perms_to_emu_perms is also added

### CODE REVIEW

**Comment by stonerhash** [comment]
Added support of SystemCodeIntegrityInformation in ZwQuerySystemInformation. This is an anti-vm / anti-analysis. The code returns value 1. Definitions in ddk.py also updated.

File: speakeasy/winenv/defs/nt/ddk.py [hunk 1]
```diff
@@ -134,7 +134,7 @@
 SYSTEM_INFORMATION_CLASS.SystemCallTimeInformation = 0x0A
 SYSTEM_INFORMATION_CLASS.SystemModuleInformation = 0x0B
 SYSTEM_INFORMATION_CLASS.SystemKernelDebuggerInformation = 0x23
-
+SYSTEM_INFORMATION_CLASS.SystemCodeIntegrityInformation =0x67
```
Comment (williballenthin) [inline speakeasy/winenv/defs/nt/ddk.py:137]:
```suggestion
SYSTEM_INFORMATION_CLASS.SystemCodeIntegrityInformation = 0x67
```

File: speakeasy/winenv/api/usermode/user32.py [hunk 2]
```diff
@@ -1311,7 +1311,8 @@ def GetKeyboardLayoutList(self, emu, argv, ctx={}):
         );
         """
         nBuff, lpList = argv
-
+        if(nBuff == 0):		# fix in case nBuff is zero and returns the number of items
+            return 1
```
Comment (williballenthin) [inline speakeasy/winenv/api/usermode/user32.py:1315]:
```suggestion
        if(nBuff == 0):
            # number of items
            return 1
```

File: speakeasy/winenv/api/kernelmode/ntoskrnl.py [hunk 3]
```diff
@@ -578,7 +600,14 @@ def ZwQuerySystemInformation(self, emu, argv, ctx={}):
                 size = len(out)
                 self.mem_write(sysinfo, out)
                 nts = ddk.STATUS_SUCCESS
-
+                
+        elif sysclass == ddk.SYSTEM_INFORMATION_CLASS.SystemCodeIntegrityInformation:
+            if sysinfo and syslen >=8:
+                class_len = (8).to_bytes(4,"little")
+                flags = (1).to_bytes(4,"little")
+                self.mem_write(sysinfo,class_len + flags)
```
Comment (williballenthin) [inline speakeasy/winenv/api/kernelmode/ntoskrnl.py:608]:
```suggestion
            if sysinfo and syslen >= 8:
                class_len = (8).to_bytes(4, "little")
                flags = (1).to_bytes(4, "little")
                self.mem_write(sysinfo, class_len + flags)
```

**Review by williballenthin — APPROVED** [review]
would you accept the formatting changes and then i'm happy to merge? thank you!

**Comment by stonerhash** [comment]
> would you accept the formatting changes and then i'm happy to merge? thank you!

Of course! Thank you for the changes

**Pull request merged**